python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -1327,7 +1329,9 @@ def get_partition_key_ranges(
 
             return self.get_ranges_for_keys(partition_keys)
 
-    def with_partition_keys(self, partition_keys: Iterable[str]) -> "DefaultPartitionsSubset":
+    def with_partition_keys(
+        self, partition_keys: Iterable[str], validate: bool = True
+    ) -> "DefaultPartitionsSubset":
```

Comment:
it's a bit odd that this isn't actually validating here given the boolean (same goes for the AllPartitionsSubset).

another option could be to branch the code in asset_backfill.py based on the type of the PartitionsDefinition, but maybe that's worse.